src/mame/namco/namcos10.cpp

```diff
@@ -3184,6 +3237,7 @@ ROM_START( gunbalina )
 
 	ROM_REGION32_LE( 0x1080000, "nand0", 0 )
 	ROM_LOAD( "gnn1a.8e", 0x0000000, 0x1080000, CRC(981b03d4) SHA1(1c55458f1b2964afe2cf4e9d84548c0699808e9f) )
+	ROM_LOAD( "gnn1a_unenc.8e", 0x0000000, 0x1080000, BAD_DUMP CRC(6c9b4d4c) SHA1(1c7fee064a03c6ae8619db9d199e2a5fd149e78d) ) // Program code is unencrypted but scrambled
```

Comment:
Do you need to load over the whole chip region, or can you use a smaller file to load over just the encrypted program area?

Reply:
Smaller files could be used but you'd need two patch files per game because the first few blocks of the program code are stored at the very end of the NAND while the rest are stored starting toward the beginning. Which would you prefer?

Reply:
It can be done with ROM_CONTINUE

Reply:
No, you can use `ROM_CONTINUE`.  Store the blocks in the file in logical order, use `ROM_LOAD` to load the first block, then `ROM_CONTINUE` for each additional contiguous chunk.

As an aside, since we know how remapping works, we should probably eventually turn everything on Namco System 10 into “canonical” Flash images as though there were no bad blocks, like we do for CV1000 where we also know how bad block remapping works.

Reply:
I'll try it with `ROM_CONTINUE` and see what I come up with.

@cuavas There's only so much that can be done in that regard. The EEP section occupies block 1 and BIOS occupies blocks 2-9. The first 9 entries of the bad block table will always point to 9 blocks of the main program code which are relocated to the reserved blocks section at the end of the NAND. Some of the dumps genuinely do have some bad blocks that could probably be fixed up but there'll always be "bad" blocks no matter what.

Reply:
Ah, I forgot that it always remaps around the settings/BIOS area.  Do you know offhand whether any of the dumps we have actually have bad blocks, apart from Point Blank 3 needing to avoid a bad block at the end of the chip?

Reply:
No, but I remember it being a not insignificant number of the dumps having at least 1 bad block. Just don't know which off the top of my head. Quick way you could check is to make a script to load 8e for all games and if at offset 0x24 you don't see 0xffffffff then you can be sure it has a bad block somewhere.

Quick script results... (not exhaustive)
gahaha
gahaha2
gjspace
gunbalina
medalnt
mrdrilrg
nflclsfb
panikuru
sekaikh
sugorotc
taiko2
taiko5
taiko6